File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/javadoc/comment/direct/Java/Go/s010698290_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	firstLine := scanner.Text()
	parts := strings.Split(firstLine, " ")
	n, _ := strconv.Atoi(parts[0])
	m, _ := strconv.Atoi(parts[1])

	solved := make([]bool, n+1)
	penal := make([]int, n+1)

	for i := 0; i < m; i++ {
		scanner.Scan()
		line := scanner.Text()
		parts := strings.Split(line, " ")
		num, _ := strconv.Atoi(parts[0])
		str := parts[1]

		if str == "A" {
			solved[num] = true
		} else {
			if !solved[num] {
				penal[num]++
			}
		}
	}

	correct := 0
	totalPenal := 0
	for problem := 1; problem <= n; problem++ {
		if solved[problem] {
			correct++
			totalPenal += penal[problem]
		}
	}

	fmt.Printf("%d %d\n", correct, totalPenal)
}

// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/mad/Java/Python/s010698290_commented.py
import sys
def main():
    data = sys.stdin.read().split()
    idx = 0
    n = int(data[idx]); idx += 1
    m = int(data[idx]); idx += 1

    solved = [False] * (n + 1)
    penalties = [0] * (n + 1)

    for _ in range(m):
        problem = int(data[idx]); idx += 1
        status = data[idx]; idx += 1

        if status[0] == 'A':
            solved[problem] = True
        else:
            if not solved[problem]:
                penalties[problem] += 1

    correct = 0
    total_penalty = 0
    for i in range(1, n+1):
        if solved[i]:
            correct += 1
            total_penalty += penalties[i]

    print(f"{correct} {total_penalty}")
if __name__ == "__main__":
    main()
# End of Code
